### ISSUE TITLE
`Course Registration`: Filter and Search Options

### ISSUE THREAD

**Issue by sebastiannikles** [ls1intum/Artemis#6185]
### Describe the solution you'd like

When registering for a new course right now there are no filter options or search bars etc. but all courses are just displayed in a list. This can look really cluttered and make it hard to find the right course.

It would be nice if I had a search bar to search for the course I like or if there were additional filter options for semester etc.

**Comment by yassinsws** [comment]
When I create courses with an admin account and then try to go to the course enrollment page with a student account, the courses do not appear. This is the case when I try it locally. But on the test servers, the courses are shown as they should on the enrollment page. 
Does anyone know about this issue or should I open a bug ticket?

**Comment by krusche** [comment]
Did you configure the enrollment in the course settings? Additionally, there might be some other conditions under which course registration is possible

**Comment by yassinsws** [comment]
Yes I did. What can the other conditions be or where can I find them? (I did not add any prerequisites for the course, I also tried adding different semesters for the course)

**Comment by solen0id** [comment]
I'm experiencing the same behaviour as @yassinsws, Courses that were created with the admin user and have the semester,  start/end date as well as the student enrollment enabled and enrollment dates configured (same start/end date as course) do not show up in the self enrollment page when logged in as a student.

**Comment by solen0id** [comment]
Oh I just found issue #6668 , will try to change the username pattern here

**Comment by yassinsws** [comment]
@solen0id you need to change in your application.yml configuration, if I remember right the pattern of the username of the students that can register for courses and make * 
I think this was resolved btw.

**Comment by solen0id** [comment]
Yeah thanks, putting an empty string as the value for `course-enrollment. allowed-username-pattern` solved it 🎉
